realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -172,10 +171,6 @@ private RealmCache(RealmConfiguration config) {
         @SuppressWarnings("unchecked")
         E realm = (E) refAndCount.localRealm.get();
 
-        // Notifies SyncPolicy that the Realm has been opened for the first time
-        if (refAndCount.globalCount == 1) {
```

Comment:
Kinda sad to see this go, but I guess we need to implement it at the OS level when we want it back.